server/server.go

```diff
@@ -235,6 +241,52 @@ func (s *Server) authenticateRequest(handler http.Handler) func(http.ResponseWri
 	}
 }
 
+// startWriteSummaries begins periodically persisting status summaries for the
+// node and its stores.
+func (s *Server) startWriteSummaries() {
+	s.stopper.RunWorker(func() {
+		ticker := time.NewTicker(s.ctx.MetricsFrequency)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				err := s.writeSummaries()
+				if err != nil {
+					log.Error(err)
+				}
+			case <-s.stopper.ShouldStop():
+				return
+			}
+		}
+	})
+}
+
+// writeSummaries retrieves status summaries from the supplied
+// NodeStatusRecorder and persists them to the cockroach data store.
+func (s *Server) writeSummaries() error {
+	nodeStatus, storeStatuses := s.recorder.GetStatusSummaries()
+	if nodeStatus != nil {
+		key := keys.NodeStatusKey(int32(nodeStatus.Desc.NodeID))
+		if err := s.db.Put(key, nodeStatus); err != nil {
```

Comment:
this `Put` and the one below may need to be guarded by `stopper.StartTask()`; i'm not sure what `db.Put` does things are shutting down.